.circleci/config.yml

```diff
@@ -0,0 +1,136 @@
+version: 2.1
+
+
+commands:
+  npm-install:
+    steps:
+      - run: sudo npm install npm@6 --global
+      - restore_cache:
+          key: js-deps-{{ checksum "package.json" }}
+      - run: npm install --no-save
+      - save_cache:
+          key: js-deps-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+
+  pip-install-docs:
+    steps:
+      - run: sudo pip install --upgrade pip
+      - restore_cache:
+          key: py-deps-{{ checksum "docs/requirements.txt" }}
+      - run: "if [ ! -d ./venv ]; then python -m venv ./venv; fi"
```

Comment:
Not sure there's much to change here, but wanted to point out that a venv created from `circleci/python:3.6` may not work with `circleci/python:3.7`, since the cache could allow older venvs to newer pythons if the Python version is updated.

Reply:
If we did something like https://github.com/apiaryio/dredd/pull/1494#discussion_r324273941 perhaps a comment around there regarding this, sure it will trip someone up in the future.